homeassistant/components/openuv/config_flow.py

```diff
@@ -38,19 +38,28 @@ def __init__(self):
 
     async def async_step_user(self, user_input=None):
         """Handle the start of the config flow."""
+        from pyopenuv.util import validate_api_key
+
         errors = {}
 
         if user_input is not None:
+            websession = aiohttp_client.async_get_clientsession(self.hass)
+            api_key_validation = await validate_api_key(
+                user_input[CONF_API_KEY], websession)
+
             identifier = '{0}, {1}'.format(
                 user_input.get(CONF_LATITUDE, self.hass.config.latitude),
                 user_input.get(CONF_LONGITUDE, self.hass.config.longitude))
 
-            if identifier not in configured_instances(self.hass):
+            if not api_key_validation:
+                errors['base'] = 'invalid_api_key'
+            elif identifier in configured_instances(self.hass):
+                errors['base'] = 'identifier_exists'
```

Comment:
It would be faster if you check this first. There won't be a need to validate an API key if this error occurs.